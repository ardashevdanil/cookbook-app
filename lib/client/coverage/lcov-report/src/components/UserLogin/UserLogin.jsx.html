<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/UserLogin/UserLogin.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/UserLogin</a> UserLogin.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>24/32</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">88.24% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>15/17</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">40% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">73.33% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>22/30</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161
162
163
164
165
166
167
168
169
170
171
172
173
174
175
176
177
178
179
180
181
182
183
184
185
186
187
188
189
190
191
192</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCheckCircle } from '@fortawesome/free-regular-svg-icons';
import PropTypes from 'prop-types';
import React from 'react';
import Dropzone from 'react-dropzone';
import './UserLogin.css';
&nbsp;
import UserLogout from '../UserLogout/UserLogout';
&nbsp;
class UserLogin extends React.Component {
  constructor() {
    super();
    this.state = {
      imgFile: null,
      login: '',
      loginPassword: '',
      signInPassword: '',
      username: '',
    };
    this.handleInputChange = this.handleInputChange.bind(this);
    this.handleLogin = this.handleLogin.bind(this);
    this.handleSignIn = this.handleSignIn.bind(this);
    this.onDrop = this.onDrop.bind(this);
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  ha</span>ndleInputChange(field, e) {
    const nextState = <span class="cstat-no" title="statement not covered" >{};</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    nextState[field] = e.target.value;</span>
<span class="cstat-no" title="statement not covered" >    this.setState(nextState);</span>
  }
&nbsp;
  handleLogin() {
    const { loginPassword, login } = { ...this.state };
    const { onLogin } = { ...this.props };
&nbsp;
    if (login === '' || loginPassword === '') return;
&nbsp;
    onLogin({
      name: login,
      password: loginPassword,
    });
    this.setState({
      login: '',
      loginPassword: '',
    });
  }
&nbsp;
  handleSignIn() {
    const { imgFile, signInPassword, username } = { ...this.state };
    const { onSignIn } = { ...this.props };
&nbsp;
    if (username === '' || signInPassword === '' || !imgFile) return;
&nbsp;
    onSignIn({
      img: imgFile,
      name: username,
      password: signInPassword,
    });
    this.setState({
      imgFile: null,
      signInPassword: '',
      username: '',
    });
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  on</span>Drop(files) {
<span class="cstat-no" title="statement not covered" >    this.setState({</span>
      imgFile: files[0],
    });
  }
&nbsp;
  render() {
    const {
      imgFile,
      login,
      loginPassword,
      signInPassword,
      username,
    } = { ...this.state };
    const {
      authorisationError,
      onLogout,
      signInError,
      user,
    } = { ...this.props };
&nbsp;
    if (user !== 'Guest') {
      return (
        &lt;UserLogout
          onLogout={onLogout}
          user={user}
        /&gt;
      );
    }
&nbsp;
    return (
      &lt;div className="UserLogin"&gt;
        &lt;div className="UserLogin__form"&gt;
          &lt;div className="UserLogin__heading"&gt;
            Login
          &lt;/div&gt;
          Login for LEMON
          &lt;input
            className="UserLogin__input"
            onChange={<span class="fstat-no" title="function not covered" >e </span>=&gt; <span class="cstat-no" title="statement not covered" >this.handleInputChange('login', e)}</span>
            placeholder="login"
            type="text"
            value={login}
          /&gt;
          &lt;input
            className="UserLogin__input"
            onChange={<span class="fstat-no" title="function not covered" >e </span>=&gt; <span class="cstat-no" title="statement not covered" >this.handleInputChange('loginPassword', e)}</span>
            placeholder="password"
            type="password"
            value={loginPassword}
          /&gt;
          &lt;div
            className={authorisationError ? <span class="branch-0 cbranch-no" title="branch not covered" >'UserLogin__error' </span>: 'UserLogin__error_hidden'}
          &gt;
            Unable to login. Check login or password.
          &lt;/div&gt;
          &lt;div
            className="UserLogin__button"
            onClick={this.handleLogin}
          &gt;
            LOGIN
          &lt;/div&gt;
          &lt;div className="UserLogin__hint"&gt;
            Dont have an account? Sign in
          &lt;/div&gt;
        &lt;/div&gt;
        &lt;div className="UserLogin__form"&gt;
          &lt;div className="UserLogin__heading"&gt;
            Sign in
          &lt;/div&gt;
          Sign in for LEMON
          &lt;input
            className="UserLogin__input"
            onChange={<span class="fstat-no" title="function not covered" >e </span>=&gt; <span class="cstat-no" title="statement not covered" >this.handleInputChange('username', e)}</span>
            placeholder="username"
            type="text"
            value={username}
          /&gt;
          &lt;input
            className="UserLogin__input"
            onChange={<span class="fstat-no" title="function not covered" >e </span>=&gt; <span class="cstat-no" title="statement not covered" >this.handleInputChange('signInPassword', e)}</span>
            placeholder="password"
            type="password"
            value={signInPassword}
          /&gt;
          &lt;Dropzone
            accept="image/jpeg, image/png"
            acceptClassName="UserLogin__dropzone_accept"
            className="UserLogin__dropzone"
            maxSize={2 * 1024 * 1024}
            onDrop={this.onDrop}
            rejectClassName="UserLogin__dropzone_reject"
          &gt;
            {!imgFile ? 'Upload a profile image' : &lt;FontAwesomeIcon icon={faCheckCircle} size="5x" /&gt;}
          &lt;/Dropzone&gt;
          &lt;div
            className={signInError ? <span class="branch-0 cbranch-no" title="branch not covered" >'UserLogin__error' </span>: 'UserLogin__error_hidden'}
          &gt;
            Username is already taken.
          &lt;/div&gt;
          &lt;div
            className="UserLogin__button"
            onClick={this.handleSignIn}
          &gt;
            SIGN UP
          &lt;/div&gt;
          &lt;div className="UserLogin__hint"&gt;
            By joing LEMON you agree to our terms of service
          &lt;/div&gt;
        &lt;/div&gt;
      &lt;/div&gt;
    );
  }
}
&nbsp;
export default UserLogin;
&nbsp;
UserLogin.propTypes = {
  authorisationError: PropTypes.bool.isRequired,
  onLogin: PropTypes.func.isRequired,
  onLogout: PropTypes.func.isRequired,
  onSignIn: PropTypes.func.isRequired,
  signInError: PropTypes.bool.isRequired,
  user: PropTypes.string.isRequired,
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Nov 11 2018 20:41:16 GMT+0300 (MSK)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
