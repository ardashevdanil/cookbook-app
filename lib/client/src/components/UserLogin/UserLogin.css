.UserLogin {
  display: flex;
  justify-content: space-around;
  margin: 85px 0;
}

.UserLogin__form {
  width: 324px;
  box-sizing: border-box;
  padding: 16px 25px;
  background-color: white;
  box-shadow: 0 0 13px 0 rgba(11, 12, 12, 0.15);
}

.UserLogin__heading {
  padding: 3px;
  margin-bottom: 29px;
  font-size: 20px;
  border-bottom: 2px solid #ddd;
  font-style: italic;
}

.UserLogin__input {
  width: 100%;
  height: 38px;
  box-sizing: border-box;
  margin-top: 20px;
  padding: 14px 20px;
  font-style: italic;
}

.UserLogin__input:focus {
  border: 1px solid #911026;
  border-radius: 3px;
  color: black;
  box-shadow: 0 0 7px 2px rgba(146, 16, 38, 0.37);
}

.UserLogin__button {
  width: 100px;
  height: 27px;
  border: solid 1px #c2c2c2;
  margin: 35px calc(50% - 50px);
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 12px; 
}

.UserLogin__button:hover {
  border-color: #911026;
  color: #911026;
  cursor: pointer;
}

.UserLogin__hint {
  text-align: center;
  font-style: italic;
  color: #9a9a9a;
  font-size: 16px;
}

.UserLogin__error {
  margin-top: 20px;
  text-align: center;
  font-style: italic;
  font-size: 16px;
  color: #911026;
}

.UserLogin__error_hidden {
  display: none;
}
@media only screen and (max-width: 1120px) {
  .UserLogin {
    margin: 0;
    display: block;
  }

  .UserLogin__form {
    width: 100%;
    padding: 5vw 7.8vw;
    box-shadow: none;
  }

  .UserLogin__heading {
    padding: 1vw;
    margin-bottom: 9vw;
    font-size: 6.3vw;
  }

  .UserLogin__input {
    height: 11.8vw;
    margin-top: 6.3vw;
    padding: 4.4vw 6.3vw;
  }

  .UserLogin__button {
    width: 31vw;
    height: 8.4vw;
    margin: 11vw calc(50% - 15.5vw);
    font-size: 3.8vw;
  }

  .UserLogin__hint {
    font-size: 5vw;
  }
}
