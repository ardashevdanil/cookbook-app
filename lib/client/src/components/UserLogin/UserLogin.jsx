import PropTypes from 'prop-types';
import React from 'react';
import './UserLogin.css';

import UserLogout from '../UserLogout/UserLogout';

class UserLogin extends React.Component {
  constructor() {
    super();
    this.state = {
      login: '',
      loginPassword: '',
      signInPassword: '',
      username: '',
    };
    this.handleInputChange = this.handleInputChange.bind(this);
    this.handleLogin = this.handleLogin.bind(this);
    this.handleSignIn = this.handleSignIn.bind(this);
  }

  handleInputChange(field, e) {
    const nextState = {};

    nextState[field] = e.target.value;
    this.setState(nextState);
  }

  handleLogin() {
    const { loginPassword, login } = { ...this.state };
    const { onLogin } = { ...this.props };

    if (login === '' || loginPassword === '') return;

    onLogin({
      name: login,
      password: loginPassword,
    });
    this.setState({
      login: '',
      loginPassword: '',
    });
  }

  handleSignIn() {
    const { signInPassword, username } = { ...this.state };
    const { onSignIn } = { ...this.props };

    if (username === '' || signInPassword === '') return;

    onSignIn({
      name: username,
      password: signInPassword,
    });
    this.setState({
      name: '',
      password: '',
    });
  }

  render() {
    const {
      login,
      loginPassword,
      signInPassword,
      username,
    } = { ...this.state };
    const {
      authorisationError,
      onLogout,
      user,
    } = { ...this.props };

    if (user !== 'Guest') {
      return (
        <UserLogout
          onLogout={onLogout}
          user={user}
        />
      );
    }

    return (
      <div className="UserLogin">
        <div className="UserLogin__form">
          <div className="UserLogin__heading">
            Login
          </div>
          Login for LEMON
          <input
            className="UserLogin__input"
            onChange={e => this.handleInputChange('login', e)}
            placeholder="login"
            type="text"
            value={login}
          />
          <input
            className="UserLogin__input"
            onChange={e => this.handleInputChange('loginPassword', e)}
            placeholder="password"
            type="text"
            value={'*'.repeat(loginPassword.length)}
          />
          <div
            className={authorisationError ? 'UserLogin__error' : 'UserLogin__error_hidden'}
          >
            Unable to login. Check login or password.
          </div>
          <div
            className="UserLogin__button"
            onClick={this.handleLogin}
          >
            LOGIN
          </div>
          <div className="UserLogin__hint">
            Dont have an account? Sign in
          </div>
        </div>
        <div className="UserLogin__form">
          <div className="UserLogin__heading">
            Sign in
          </div>
          Sign in for LEMON
          <input
            className="UserLogin__input"
            onChange={e => this.handleInputChange('username', e)}
            placeholder="username"
            type="text"
            value={username}
          />
          <input
            className="UserLogin__input"
            onChange={e => this.handleInputChange('signInPassword', e)}
            placeholder="password"
            type="text"
            value={'*'.repeat(signInPassword.length)}
          />
          <div
            className="UserLogin__button"
            onClick={this.handleSignIn}
          >
            SIGN UP
          </div>
          <div className="UserLogin__hint">
            By joing LEMON you agree to our terms of service
          </div>
        </div>
      </div>
    );
  }
}

export default UserLogin;

UserLogin.propTypes = {
  authorisationError: PropTypes.bool.isRequired,
  onLogin: PropTypes.func.isRequired,
  onLogout: PropTypes.func.isRequired,
  onSignIn: PropTypes.func.isRequired,
  user: PropTypes.string.isRequired,
};
