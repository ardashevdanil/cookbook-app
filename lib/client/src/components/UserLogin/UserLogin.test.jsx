import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

import UserLogin from './UserLogin';

Enzyme.configure({ adapter: new Adapter() });

describe('test UserLogin component', () => {
  const onLogin = jest.fn();
  const onLogout = jest.fn();
  const onSignIn = jest.fn();
  const component = Enzyme.shallow(
    <UserLogin
      authorisationError={false}
      onLogin={onLogin}
      onLogout={onLogout}
      onSignIn={onSignIn}
      user={'Guest'}
    />,
  );

  test('renders correctly', () => {
    expect(component.debug()).toMatchSnapshot();
  });

  test('doesn\'t call onLogin after click on login button with empty state', () => {
    component.find('.UserLogin__button').at(0).simulate('click');

    expect(onLogin.mock.calls.length).toBe(0);
  });

  test('calls onLogin after click on login button after inserting login and password in state', () => {
    component.setState({
      login: 'test',
      loginPassword: 'test',
    });
    component.find('.UserLogin__button').at(0).simulate('click');

    expect(onLogin.mock.calls[0][0]).toEqual({
      name: 'test',
      password: 'test',
    });
  });

  test('doesn\'t call onSignIn after click on signin button with empty state', () => {
    component.find('.UserLogin__button').at(1).simulate('click');

    expect(onSignIn.mock.calls.length).toBe(0);
  });

  test('calls onSignIn after click on signin button after inserting name and password in state', () => {
    component.setState({
      username: 'test',
      signInPassword: 'test',
    });
    component.find('.UserLogin__button').at(1).simulate('click');

    expect(onSignIn.mock.calls[0][0]).toEqual({
      name: 'test',
      password: 'test',
    });
  });

  test('should render logout window than user != Guest', () => {
    component.setProps({ user: 'Vasya' });

    expect(component.children.length).toBe(1);
  });
});
