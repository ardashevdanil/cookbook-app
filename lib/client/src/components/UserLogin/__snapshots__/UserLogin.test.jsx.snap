// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test UserLogin component renders correctly 1`] = `
"<div className=\\"UserLogin\\">
  <div className=\\"UserLogin__form\\">
    <div className=\\"UserLogin__heading\\">
      Login
    </div>
    Login for LEMON
    <input className=\\"UserLogin__input\\" onChange={[Function: onChange]} placeholder=\\"login\\" type=\\"text\\" value=\\"\\" />
    <input className=\\"UserLogin__input\\" onChange={[Function: onChange]} placeholder=\\"password\\" type=\\"text\\" value=\\"\\" />
    <div className=\\"UserLogin__error_hidden\\">
      Unable to login. Check login or password.
    </div>
    <div className=\\"UserLogin__button\\" onClick={[Function: bound handleLogin]}>
      LOGIN
    </div>
    <div className=\\"UserLogin__hint\\">
      Dont have an account? Sign in
    </div>
  </div>
  <div className=\\"UserLogin__form\\">
    <div className=\\"UserLogin__heading\\">
      Sign in
    </div>
    Sign in for LEMON
    <input className=\\"UserLogin__input\\" onChange={[Function: onChange]} placeholder=\\"username\\" type=\\"text\\" value=\\"\\" />
    <input className=\\"UserLogin__input\\" onChange={[Function: onChange]} placeholder=\\"password\\" type=\\"text\\" value=\\"\\" />
    <div className=\\"UserLogin__button\\" onClick={[Function: bound handleSignIn]}>
      SIGN UP
    </div>
    <div className=\\"UserLogin__hint\\">
      By joing LEMON you agree to our terms of service
    </div>
  </div>
</div>"
`;
